### ISSUE TITLE
Ignoring tracker errors and viewing tracker status

### ISSUE THREAD

**Issue by Avamander** [jfurrow/flood#486]
Is there a way to make Flood ignore tracker errors, it's so common for some tracker to be down and then everything just has a red bar making it rather difficult to have a good overview. It's also rather difficult to find out an individual tracker's status, would it be possible to display that too?

**Comment by sachaw** [comment]
+1 for better error display

**Comment by jfurrow** [comment]
I'll consider this.

Where in the UI would you like to find an individual tracker's status?

**Comment by sachaw** [comment]
I think on the tracker page, possibly highlight the one with an error and either have an arrow to expand the error, have inline or newline right below it

**Comment by Avamander** [comment]
@jfurrow I agree with @sachaw, but will you consider both changes? Because pretty much every torrent of mine is painted red at the moment :D

**Comment by randomnonsense** [comment]
@jfurrow Hiding offline trackers and hiding actual tracker errors are two separate things. Perhaps the best way to go would be marking offline trackers "hidden" or whatever of the sort.

Trackers report errors to clients for a very good reason. This must not be hidden.

**Comment by sachaw** [comment]
I thhink optimising for readability of the list:
Grey if stopped
Green if seeding
Black if done and not seeding
If there are errors, still default to whatever state it would be in otherwise
e.g. one tracker failed but is still seeding, it should be green.

Possibly just leave the exclamation mark where it is so you can still see what torrents have issues.

**Comment by randomnonsense** [comment]
What @sachaw proposes makes sense.

Since we are at this, @jfurrow could also consider breaking current "Percent complete" into two separate columns, "Status" and "Percent complete". This should allow easy sorting via status within particular category simply by clicking column headers.

Also a better readable "percent complete" bar would be welcome, e.g. something with borders easily visible (so it can be seen easily how much is done), for example:

![bar](https://user-images.githubusercontent.com/31191881/31857551-841caa38-b6e1-11e7-9990-70a03275f045.png)

**Comment by sachaw** [comment]
sorry @randomnonsense That's what I meant.

**Comment by jfurrow** [comment]
I agree with what @sachaw suggested — leaving the colors of the torrent alone, but displaying the exclamation icon when an error is detected.

Discerning between tracker errors and timeouts is not possible (at least not via rtorrent), as far as I know. Flood considers a torrent to have an error if the `message` property on the tracker contains a string. If it does, then Flood determines the torrent is in the error state (this is consistent with ruTorrent's behavior, as far as I know). I don't think there's any way to tell if the tracker is timing out, or in some other state.

@randomnonsense I'd like to find another solution for better legibility in the progress bar. Maybe opening a separate ticket for that would help facilitate further discussion on a better design.

**Comment by sachaw** [comment]
#500

**Comment by sachaw** [comment]
#500

**Comment by Avamander** [comment]
I'm mainly wanting the option to ignore trackers because public trackers are god awful, they randomly go down, they give funky errors etc. so my list of Linux isos is basically all red because even with two online trackers one is always down.

**Comment by randomnonsense** [comment]
@jfurrow can you make this user selectable? E.g. "Hide tracker errors", as a tick-box in Settings. Or since rtorrent (and hence flood) can read private flags, you could go for "Hide tracker errors on non-private torrents". This way anybody can choose whatever they want.

**Comment by LFlare** [comment]
I'm going to casually bump this because I think this is kind of a pressing issue when trying to view individual torrent statuses. As I speak, my entire setup is filled with red torrents that are very clearly downloading and not actually fatal unlike the impression that the red error is giving.
